File: frontend/src/features/admin/services/bus-type-service.ts
import axiosClient from "@/services/http/axios";
import { BusType, SeatMapItem } from "../types";

interface ApiListResponse<T> {
    code: number;
    message: string;
    result: T[];
}

interface ApiSingleResponse<T> {
    code: number;
    message: string;
    result: T;
}

export interface BusTypeRequest {
    name: string;
    totalSeats: number;
    seatMap: SeatMapItem[];
}

export const busTypeService = {
    getAll: async (): Promise<BusType[]> => {
        const { data } = await axiosClient.get<ApiListResponse<BusType>>("/fleet/bus-types");
        return data.result;
    },

    getById: async (id: number): Promise<BusType> => {
        const { data } = await axiosClient.get<ApiSingleResponse<BusType>>(`/fleet/bus-types/${id}`);
        return data.result;
    },

    create: async (request: BusTypeRequest): Promise<BusType> => {
        const { data } = await axiosClient.post<ApiSingleResponse<BusType>>("/fleet/bus-types", request);
        return data.result;
    },

    update: async (id: number, request: BusTypeRequest): Promise<BusType> => {
        const { data } = await axiosClient.put<ApiSingleResponse<BusType>>(`/fleet/bus-types/${id}`, request);
        return data.result;
    },

    delete: async (id: number): Promise<void> => {
        await axiosClient.delete(`/fleet/bus-types/${id}`);
    }
};

File: backend/src/main/java/com/bus/system/modules/fleet/service/BusTypeService.java
package com.bus.system.modules.fleet.service;

import com.bus.system.modules.fleet.dto.request.BusTypeRequest;
import com.bus.system.modules.fleet.dto.response.BusTypeResponse;

import java.util.List;

public interface BusTypeService {
    BusTypeResponse create(BusTypeRequest request);

    BusTypeResponse update(Long id, BusTypeRequest request);

    void delete(Long id);

    void restore(Long id);

    List<BusTypeResponse> getAll();

    List<BusTypeResponse> getDeletedBusTypes();

    BusTypeResponse getById(Long id);
}
